src/operations/InlayHints.cpp

```diff
@@ -157,74 +224,32 @@ struct InlayHintVisitor : public Luau::AstVisitor
         return true;
     }
 
-    bool visit(Luau::AstExprFunction* func) override
+    bool visit(Luau::AstStatLocalFunction* funcStat) override
     {
+        auto func = funcStat->func;
         auto ty = module->astTypes.find(func);
         if (!ty)
             return false;
 
-        auto followedTy = Luau::follow(*ty);
-        if (auto ftv = Luau::get<Luau::FunctionType>(followedTy))
-        {
-            // Add return type annotation
-            if (config.inlayHints.functionReturnTypes)
-            {
-                if (!func->returnAnnotation && func->argLocation && !isNoOpFunction(func))
-                {
-                    lsp::InlayHint hint;
-                    hint.kind = lsp::InlayHintKind::Type;
-                    hint.label = ": " + types::toStringReturnType(ftv->retTypes, stringOptions);
-                    hint.position = textDocument->convertPosition(func->argLocation->end);
-                    makeInsertable(config, hint, ftv->retTypes);
-                    hints.emplace_back(hint);
-                }
-            }
+        inlayHintForFuncType(func, ty);
 
-            // Parameter types hint
-            if (config.inlayHints.parameterTypes)
-            {
-                auto it = Luau::begin(ftv->argTypes);
-                if (it != Luau::end(ftv->argTypes))
-                {
-                    // Skip first item if it is self
-                    if (func->self && isMethod(ftv))
-                        it++;
+        return true;
+    }
 
-                    for (auto param : func->args)
-                    {
-                        if (it == Luau::end(ftv->argTypes))
-                            break;
+    bool visit(Luau::AstStatFunction* funcStat) override
+    {
+        auto func = funcStat->func;
+        Luau::AstExpr* lookupType = func;
 
-                        auto argType = *it;
-                        if (!param->annotation && param->name != "_")
-                        {
-                            lsp::InlayHint hint;
-                            hint.kind = lsp::InlayHintKind::Type;
-                            hint.label = ": " + Luau::toString(argType, stringOptions);
-                            hint.position = textDocument->convertPosition(param->location.end);
-                            makeInsertable(config, hint, argType);
-                            hints.emplace_back(hint);
-                        }
+        // TODO: Remove me later if this is in fact not an intended feature in the new solver
```

Comment:
Can you explain what is happening here? I don't get why this changed

Reply:
Essentially, in the new solver, the astType for the AstStatFunction is incomplete with both arguments typed as `unknown` if that function is a member function of an object (e.g. `table:test()`). However, the complete type is accessible using the AstExprIndexName ast type.

See new solver AssignPropConstraint and my OSS discord discussion
![image](https://github.com/user-attachments/assets/090b3614-28de-448d-8ca2-636b46ccf394)

Reply:
Also the git diff is HORRIBLE here I recommend reading the actual file to reduce confusion if you aren't

Reply:
I think we should also take a deeper look at Luau here before fixing. This either seems like a Luau regression, or there must be a new struct to search this data up instead of astTypes.

I'll take a look if I'm free if you don't get round to it

Reply:
The other alternative is this is stored in a refinement, which I think is the correct place to look instead? See scope->rvalueRefinements

Reply:
Can you move this change out of this PR and into a separate one? I think we need to check what's up with Luau here first before changing it, but I don't want to block the rest of the fixes

Reply:
Split off into #815